lib/pbench/server/database/models/active_token.py

```diff
@@ -0,0 +1,56 @@
+from datetime import datetime
+from typing import TypeVar
+
+from sqlalchemy import Column, DateTime, ForeignKey, Integer, String
+from sqlalchemy.orm import relationship
+
+from pbench.server.database.database import Database
+
+AT = TypeVar("AT", bound="ActiveToken")
+
+
+class ActiveToken(Database.Base):
+    """Token model for storing the active auth tokens.
+
+    Each token is associated with a given User object, and stores its
+    expiration time.
+    """
+
+    # Table name is plural so it looks better SQL statements.
+    __tablename__ = "active_tokens"
+    id = Column(Integer, primary_key=True, autoincrement=True)
+    token = Column(String(500), unique=True, nullable=False, index=True)
+    expiration = Column(DateTime, nullable=False, index=True)
+    user_id = Column(
+        Integer,
+        ForeignKey("users.id", ondelete="CASCADE"),
+        nullable=False,
+        # no need to add index=True, all FKs have indexes
+    )
```

Comment:
Are we sure we want `index=True` on the `expiration` column?  We're never going to look up a user or token by expiration, right?  (@dbutenhof?)

Reply:
Pete added code to search for expired tokens. Whether that adds enough CPU overhead to justify the space overhead of indexing the column is a valid question, but short of a performance analysis I don't see a problem.